packages/office-ui-fabric-react/src/components/Tooltip/TooltipHost.tsx

```diff
@@ -114,24 +114,26 @@ export class TooltipHost extends BaseComponent<ITooltipHostProps, ITooltipHostSt
       }
     }
 
-    this.setState({
-      isTooltipVisible: true
-    });
+    this.toggleTooltip(true);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipMouseLeave(ev: any) {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
   }
 
   // Hide Tooltip
   @autobind
   private _onTooltipCallOutDismiss() {
-    this.setState({
-      isTooltipVisible: false
-    });
+    this.toggleTooltip(false);
+  }
+
+  private toggleTooltip(isTooltipVisible: boolean) {
+    if (this.props.onTooltipToggle && this.state.isTooltipVisible !== isTooltipVisible) {
+      this.props.onTooltipToggle(isTooltipVisible);
+    }
```

Comment:
That's a good point, and I agree on that it "can potentially skip".
But I'm not confident that using setState callback will be safer. My point is that we would need this before calling `setState`:
```
const hasChanged = this.state.isTooltipVisible !== isTooltipVisible;
```
And that `state` could be stale, and it's commanding what we will do in the callback. So after thinking further, IMO the safest is to **remove the changed checking** and trigger always. I think so because React is optimized to swallow multiple no-changes, so extra calls should not have any impact in performance.

As a side note, see the [onChanged from Dropdown](https://github.com/OfficeDev/office-ui-fabric-react/blob/6ea835ec7c1324dee1e8fbf1e212b44beee286bd/packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx#L252): it doesn't even call setState if indexes are the same, which is a risk because it's reading `this.state` which can be potentially stale.